web/src/main/java/org/cbioportal/web/ClinicalDataController.java

```diff
@@ -13,54 +24,125 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @Api(tags = "Clinical Data", description = " ")
 public class ClinicalDataController {
 
-    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET)
-    @ApiOperation("Get all clinical data in a sample in a study")
-    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataInSampleInStudy(@PathVariable String studyId,
-                                                                                                 @PathVariable String sampleId,
-                                                                                                 @RequestParam(required = false) String attributeId,
-                                                                                                 @RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber) {
+    @Autowired
+    private ClinicalDataService clinicalDataService;
 
-        throw new UnsupportedOperationException();
+    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all clinical data of a sample in a study")
+    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataOfSampleInStudy(
+            @PathVariable String studyId,
+            @PathVariable String sampleId,
+            @RequestParam(required = false) String attributeId,
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
+            @RequestParam(required = false) ClinicalDataSort sort,
+            @RequestParam(defaultValue = "ASC") Direction direction) {
```

Comment:
It looks like this may be SQL specific. If this parameter gets inlined into the mybatis select statement, I think it would be wise to insert one level of indirection and not make it a SQL keyword. (open to discussion obviously)
I see below that this maps to a Direction Enum rather than direct to the mybatis level, so ok.